src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -317,6 +317,22 @@ internal static Syntax.InternalSyntax.DirectiveStack ApplyDirectives(this Syntax
             return builder.ToList();
         }
 
+        #region ITypeSymbol
+
+        /// <summary>
+        /// Returns true iff the supplied type is sbyte, byte, short, ushort, int, uint,
+        /// long, ulong, bool, char, string, or an enum-type, or if it is the nullable type
+        /// corresponding to one of those types. These types were permitted as the governing
+        /// type of a switch statement in C# 6.
+        /// </summary>
+        public static bool IsValidV6SwitchGoverningType(this ITypeSymbol type)
```

Comment:
Would this be more appropriate to expose on some more specialized type, perhaps on SemanticModel? Also it would be nice to avoid ```V6``` in the name. Perhaps ```IsPrimitiveSwitchType```?